quaternions/quaternions.py

```diff
@@ -293,21 +293,75 @@ def from_qmethod(source, target, probabilities=None):
         return Quaternion._first_eigenvector(K)
 
     @staticmethod
-    def average(*quaternions, weights=None):
+    def average(*quaternions, weights=()):
         '''
         Return the quaternion such that its matrix minimizes the square distance
         to the matrices of the quaternions in the argument list.
 
         See Averaging Quaternions, by Markley, Cheng, Crassidis, Oschman.
         '''
         B = np.array([q.coordinates for q in quaternions])
-        if not weights:
+        if not any(True for _ in weights):
```

Comment:
Why this instead of the empty list/tuple?

Reply:
I'll change it to: use  None as default value for weights, and then check `if weights is None`.